### ISSUE TITLE
In the developer guide, forbid new global reactor unit tests

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5371]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#5371|
|Type|enhancement|
|Created|2011-11-07 15:46:02Z|
|Branch|https://github.com/twisted/twisted/tree/no-global-reactor-testing-5371|

On the mailing list, <http://twistedmatrix.com/pipermail/twisted-python/2011-November/024732.html>, proposed the idea that we shouldn't be adding _new_ unit tests to Twisted which use the global reactor.  No disagreement in the resulting discussion.

<details><summary>Searchable metadata</summary>

```
trac-id__5371 5371
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_no_global_reactor_testing_5371 branches/no-global-reactor-testing-5371
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1320680762000000 1320680762000000
changetime__1321210195000000 1321210195000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [33092]) Branching to 'no-global-reactor-testing-5371'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="frap's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|frap set status to `closed`|
|-|-|

Documentation looks good and understandable. no formatiing errors nor spellling mistakes in browser. Makes sense to me.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_4"></a>|@itamarst set status to `reopened`|
|-|-|

When reviewing branches, if you're done with the review you should remove the "review" keyword and reassign back to author, not close the ticket.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_5"></a>|@thijstriemstra set owner to @exarkun<br>@thijstriemstra set status to `new`|
|-|-|

Looks good to merge.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_6"></a>|@itamarst commented|
|-|-|

There's a spelling mistake: "unweildy" instead of "unwieldy". Also maybe change the title from "Tips" to "Guidelines".

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun commented|
|-|-|

(In [33115]) incorporate itamar's suggested improvements

refs [#5371](https://github.com/twisted/twisted/issues/5371)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun set status to `closed`|
|-|-|

(In [33116]) Merge no-global-reactor-testing-5371 

Author: exarkun
Reviewer: frap, thijs, itamar
Fixes: [#5371](https://github.com/twisted/twisted/issues/5371)

Codify the proscription on global reactor use in unit tests.
